src/NServiceBus.AcceptanceTests/Core/AutomaticSubscriptions/When_starting_an_endpoint_with_autosubscribe_disabled.cs

```diff
@@ -0,0 +1,129 @@
+﻿namespace NServiceBus.AcceptanceTests.Core.AutomaticSubscriptions
+{
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using Features;
+    using NUnit.Framework;
+
+    public class When_starting_an_endpoint_with_autosubscribe_disabled : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_not_autosubscribe_any_events_on_native_pubsub()
+        {
+            Requires.NativePubSubSupport();
+
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<Subscriber>(e => e
+                    .When(async (session, ctx) =>
+                    {
+                        await session.Subscribe<ManuallySubscribedEvent>();
+                        ctx.SubscribedToEvent = true;
+                    }))
+                .WithEndpoint<Publisher>(e => e
+                    .When(
+                        ctx => ctx.SubscribedToEvent,
+                        async session =>
+                        {
+                            await session.Publish(new NonSubscribedEvent());
+                            await session.Publish(new ManuallySubscribedEvent());
+                        }))
+                .Done(ctx => ctx.ManuallySubscribedEventReceived)
+                .Run();
+
+            Assert.IsTrue(context.ManuallySubscribedEventReceived);
+            Assert.IsFalse(context.NonSubscribedEventReceived);
```

Comment:
This assertion can give false positives when `ManuallySubscribedEventReceived` and `NonSubscribedEventReceived` are both subscribed to but `NonSubscribedEventReceived` happens to be received first.
Could we make the test more predictable by testing if the `Subscribe` operations are called or not for the events by the `AutoSubscribe` feature?